internal/nostr/nostr.go

```diff
@@ -13,53 +13,42 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-const (
-	NIP_47_INFO_EVENT_KIND = 13194
-	NIP_47_REQUEST_KIND    = 23194
-	NIP_47_RESPONSE_KIND   = 23195
-)
-
-type WalletConnectInfo struct {
-	RelayURL     string
-	WalletPubkey string
-	Secret       string
-}
-
-type ErrorResponse struct {
-	Message string `json:"message"`
+type Config struct {
+	DefaultRelayURL string
 }
 
-type InfoRequest struct {
-	RelayURL     string `json:"relayUrl"`
-	WalletPubkey string `json:"walletPubkey"`
+type NostrService struct {
+	config      *Config
+	defaultRelay *nostr.Relay
 }
 
-type NIP47Request struct {
-	RelayURL     string       `json:"relayUrl"`
-	WalletPubkey string       `json:"walletPubkey"`
-	SignedEvent  *nostr.Event `json:"event"`
-	WebhookURL   string       `json:"webhookURL"`
-}
+func NewNostrService(config *Config) (*NostrService, error) {
+	logrus.Info("connecting to the relay...")
+	relay, err := nostr.RelayConnect(context.Background(), config.DefaultRelayURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to default relay: %w", err)
+	}
 
-func handleError(w http.ResponseWriter, err error, message string, httpStatusCode int) {
-	logrus.WithError(err).Error(message)
-	http.Error(w, message, httpStatusCode)
+	return &NostrService{
+		config:      config,
+		defaultRelay: relay,
+	}, nil
 }
 
-func InfoHandler(c echo.Context) error {
+func (svc *NostrService) InfoHandler(c echo.Context) error {
 	var requestData InfoRequest
 	if err := c.Bind(&requestData); err != nil {
 		return c.JSON(http.StatusBadRequest, ErrorResponse{
 			Message: fmt.Sprintf("error decoding info request: %s", err.Error()),
 		})
 	}
 
-	logrus.Info("connecting to the relay...")
-	relay, err := nostr.RelayConnect(c.Request().Context(), requestData.RelayURL)
+	relay, isCustomRelay, err := svc.getRelayConnection(c.Request().Context(), requestData.RelayURL)
 	if err != nil {
-		return c.JSON(http.StatusBadRequest, ErrorResponse{
-			Message: fmt.Sprintf("error connecting to relay: %s", err.Error()),
-		})
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("error connecting to custom relay: %s", err))
+	}
+	if isCustomRelay {
+		defer relay.Close()
```

Comment:
did we miss closing the relay connection before?

Reply:
😬 yes